pysbd/abbreviation_replacer.py

```diff
@@ -71,26 +71,26 @@ def replace_period_of_abbr(self, txt, abbr):
         txt = txt[1:]
         return txt
 
-
     def search_for_abbreviations_in_string(self):
         original = self.text
-        lowered = original.lower()
-        for abbr in self.lang.Abbreviation.ABBREVIATIONS:
-            stripped = abbr.strip()
-            if stripped not in lowered:
-                continue
-            abbrev_match = re.findall(
-                r"(?:^|\s|\r|\n){}".format(stripped), original, flags=re.IGNORECASE
-            )
-            if not abbrev_match:
-                continue
-            next_word_start = r"(?<={" + str(re.escape(stripped)) + "} ).{1}"
-            char_array = re.findall(next_word_start, self.text)
-            for ind, match in enumerate(abbrev_match):
-                self.text = self.scan_for_replacements(
-                    self.text, match, ind, char_array
-                )
-        return self.text
+        abbrs = "|".join([re.escape(abr.strip()) for abr in self.lang.Abbreviation.ABBREVIATIONS])
+        abbregex = re.compile(r"(?:^|\s|\r|\n)({})\b".format(abbrs), flags=re.IGNORECASE)
+        abbrev_matches = re.findall(abbregex, original)
+        try:
```

Comment:
Why is this try except now needed?

Reply:
Earlier for-loop on each abbreviation logic from pragmatic-segmenter was too time-taking hence we did refactoring of this function to improving performance on bigger text files. Languages - `ru, it, bg` - was breaking due to the regex approach. Hence, added `ABBREVIATIONS2` to handle those
languages specific abbreviation cases.